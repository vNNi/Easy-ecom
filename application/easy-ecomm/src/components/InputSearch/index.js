export { default } from './input-search';
