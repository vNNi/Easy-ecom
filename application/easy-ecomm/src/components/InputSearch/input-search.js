import { Input } from 'antd';

import { Container } from './input-search.styles';

const InputSearch = ({ onSearch, ...props }) => {
  return (
    <Container>
      <Input.Search
        placeholder="input search text"
        onSearch={onSearch}
        enterButton
        {...props}
      />
    </Container>
  );
};

export default InputSearch;
