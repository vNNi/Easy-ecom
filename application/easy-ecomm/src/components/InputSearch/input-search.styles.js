import styled from 'styled-components';

export const Container = styled.div`
  .ant-btn-primary {
    background-color: ${({ theme }) => theme.colors.primary};
  }
`;
