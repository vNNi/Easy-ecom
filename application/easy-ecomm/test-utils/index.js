import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import '@testing-library/jest-dom/extend-expect'; // jest-dom custom assertions
import 'jest-styled-components';

import Theme from '../src/styles/theme';

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};

export const renderWithTheme = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
