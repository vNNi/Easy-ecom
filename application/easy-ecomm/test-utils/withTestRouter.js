import React from 'react';
import { RouterContext } from 'next/dist/next-server/lib/router-context';

export default function withTestRouter(tree, router = {}) {
  const {
    route = '',
    pathname = '',
    pagename = '',
    pagetype = '',
    query = {},
    asPath = '',
    push = async () => true,
    replace = async () => true,
    reload = () => null,
    back = () => null,
    prefetch = async () => undefined,
    beforePopState = () => null,
    isFallback = false,
    events = {
      emit: () => null,
      off: () => null,
      on: () => null,
    },
  } = router;

  return (
    <RouterContext.Provider
      value={{
        asPath,
        back,
        beforePopState,
        events,
        isFallback,
        pagename,
        pagetype,
        pathname,
        prefetch,
        push,
        query,
        reload,
        replace,
        route,
      }}
    >
      {tree}
    </RouterContext.Provider>
  );
}
