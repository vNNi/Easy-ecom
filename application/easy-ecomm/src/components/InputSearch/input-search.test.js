import { renderWithTheme } from '../../../test-utils';

import InputSearch from './input-search';

describe('[Components] - input search', () => {
  test('should render correctly', () => {
    const comp = renderWithTheme(<InputSearch />);
    expect(comp).toMatchSnapshot();
  });
});
