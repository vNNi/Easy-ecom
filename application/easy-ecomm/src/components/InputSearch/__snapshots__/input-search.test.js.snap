// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Components] - input search should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 .ant-btn-primary {
  background-color: #0077b6;
  border-color: #0077b6;
}

<body>
    <div>
      <div
        class="c0"
      >
        <span
          class="ant-input-group-wrapper ant-input-search ant-input-search-with-button"
        >
          <span
            class="ant-input-wrapper ant-input-group"
          >
            <input
              class="ant-input"
              placeholder="input search text"
              type="text"
              value=""
            />
            <span
              class="ant-input-group-addon"
            >
              <button
                class="ant-btn ant-btn-primary ant-input-search-button"
                type="button"
              >
                <span
                  aria-label="search"
                  class="anticon anticon-search"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="search"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                    />
                  </svg>
                </span>
              </button>
            </span>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="input-searchstyles__Container-f8f9gh-0 idPiTg"
    >
      <span
        class="ant-input-group-wrapper ant-input-search ant-input-search-with-button"
      >
        <span
          class="ant-input-wrapper ant-input-group"
        >
          <input
            class="ant-input"
            placeholder="input search text"
            type="text"
            value=""
          />
          <span
            class="ant-input-group-addon"
          >
            <button
              class="ant-btn ant-btn-primary ant-input-search-button"
              type="button"
            >
              <span
                aria-label="search"
                class="anticon anticon-search"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </span>
            </button>
          </span>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
